---
title: Fedora20にしたらgraphiteが動かなくなった。
author: kongou_ae
layout: post
date: 2014-08-05
url: /blog/archives/2124
categories:
  - python
  - Server
---
　Fedora19にyumでインストールしたgraphiteが、Fedora20にアップグレードした時から動かなくなりました。具体的な症状は、WEBでアクセスすると以下のメッセージとともにInternal Server errorになります。

　モジュールがimport errorになる理由は、[Django import error &#8211; no module named django.conf.urls.defaults][1]に記載のある通り、Django1.6からdefaultモジュールが廃止されているためのようです。確かに、Fedora19のpython-djangoは1.5.8、Fedora20のpytho-djangoは1.6.5です。

　Fedora20のyumでインストールできるgraphite-webパッケージはこの変更に追随していないようなので、githubのコミットを元に自分でパッチを当てました。なお、作業中のvimでは以下の2点を手修正しています。

  1. manage.pyへのパッチが失敗したので、コメントアウト
  2. ファイルのパスが本番環境と異なるので、正しいパスに置換

　パッチ適応後、Internal server errorは解消され、正常にgraphiteの管理画面が表示されるようになりました。

 [1]: http://stackoverflow.com/questions/19962736/django-import-error-no-module-named-django-conf-urls-defaults